pkg/kev/sources.go

```diff
@@ -33,7 +34,34 @@ func (s *Sources) CalculateBaseOverride(opts ...BaseOverrideOpts) error {
 		return errors.Errorf("%s\nsee compose files: %v", err.Error(), s.Files)
 	}
 
-	s.override = extractLabels(ready)
+	s.override = &composeOverride{
+		Version: ready.version,
+	}
+
+	for _, svc := range ready.Services {
+		target := ServiceConfig{
+			Name:       svc.Name,
+			Labels:     map[string]string{},
+			Extensions: svc.Extensions,
+		}
+
+		setDefaultLabels(&target)
+		extractVolumesLabels(ready, s.override)
+		extractLabels(svc, &target)
+
+		k8sConf, err := config.K8SCfgFromMap(svc.Extensions)
```

Comment:
We now bite into the extensions so would be good to update the function description a little bit, to reflect that perhaps?

NOTE: I review on assumption that this change is backward compatible with labels driven config.

Reply:
updated